src/callbacks/monitor.py

```diff
@@ -77,10 +80,19 @@ def state_dict(self):
         }
 
     def load_state_dict(self, state_dict):
-        self.checkpoints_dir = state_dict['checkpoints_dir']
-        self.mode = state_dict['mode']
-        self.target = state_dict['target']
-        self.saved_freq = state_dict['saved_freq']
-        self.early_stop = state_dict['early_stop']
-        self.best = state_dict['best']
-        self.not_improved_count = state_dict['not_improved_count']
+        if self.mode == state_dict['mode'] and self.target == state_dict['target']:
+            self.best = state_dict['best']
+            self.not_improved_count = state_dict['not_improved_count']
+        else:
+            LOGGER.warning(f"The mode and target are changed from "
+                           f"{state_dict['mode']} {state_dict['target']} to {self.mode} {self.target}.")
+
+        if self.saved_freq != state_dict['saved_freq']:
+            LOGGER.warning(f"The saved_freq is changed from {state_dict['saved_freq']} to {self.saved_freq}.")
+
+        if self.early_stop != state_dict['early_stop']:
+            LOGGER.warning(f"The early_stop is changed from {state_dict['early_stop']} to {self.early_stop}.")
+
+        if self.not_improved_count >= self.early_stop:
+            LOGGER.critical(f"Load the checkpoint that should have to be early stopped.")
+            sys.exit()
```

Comment:
Should we need to stop the process or just throw a warning message?

Reply:
I think `sys.exit()` will stop the process.

Reply:
I see. But I mean whether it's necessary to stop the process. Maybe somebody needs to resume the training process after the `early stop` break?

Reply:
I think the logic of `early_stop` is that the users want to stop the training process (with certain setting) early. They could still resume the training process as `self.mode` or `self.target` has changed (`self.not_improved_count` would be reset to `np.inf`). But if not the case, I think we should prevent the users from resuming.